internal/neofs/neofs.go

```diff
@@ -316,8 +331,14 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 		return oid.ID{}, fmt.Errorf("save object via connection pool: %w", err)
 	}
 
-	chunk := make([]byte, x.cfg.MaxObjectSize)
-	_, err = io.CopyBuffer(writer, prm.Payload, chunk)
+	data := x.buffers.Get()
+	chunk, _ := data.(*[]byte)
+
+	defer func() {
```

Comment:
Can be a bit more efficient if you're to do it after `CopyBuffer`, you don't need to wait for `Close()` to return the buffer to the pool.